frontend/src/Components/MyTasks/DeliveriesContainer.tsx

```diff
@@ -148,23 +26,56 @@ const DeliveriesContainer = (props: {
     });
   };
 
-  // filtering logic
-  const dateFilteredTasks = dummyTasks.filter(
-    (task) => formatDate(task.deliveryTime) == props.dateFilter
-  );
-  let filteredTasks = dateFilteredTasks; // ALLTASKS filter
-  if (props.completionFilter === "COMPLETED") {
-    // if filter is set as COMPLETED, apply filter
-    filteredTasks = dateFilteredTasks.filter((task) => task.isCompleted);
-  } else if (props.completionFilter === "UPCOMING") {
-    filteredTasks = dateFilteredTasks.filter((task) => !task.isCompleted);
+  // function passed to sort function to sort deliveries based on routePosition
+  const compareDeliveries = (delivery1 : MealDeliveryInterface, delivery2 : MealDeliveryInterface) => {
+    if (delivery1.routePosition < delivery2.routePosition) {
+      return -1;
+    } else if (delivery1.routePosition > delivery2.routePosition) {
+      return 1;
+    }
+    return 0;
+  } 
+
+  let oneDayTask: TaskInterface | null = null; // will be selected date's task
+
+   // filtering based on date 
+  if (fetchedTasks) {
+    console.log("in deliveriesContainter ", fetchedTasks); //fetchedTasks.tasks
+    // we now assume that there is only one task associated to one date.
+    for (let task of fetchedTasks) {
+      console.log("date filtering now: ", formatDate(new Date(task.date)), props.dateFilter)
+      if (task.date && formatDate(new Date(task.date)) == props.dateFilter) {
+        oneDayTask = task; // filter the task
+        break;
+      } 
+    }
   }
 
+  let filteredDeliveries: MealDeliveryInterface[] = [];
+
+  // filtering based on completion
+  if (oneDayTask) { // only if there is a task assigned for the current day
+    filteredDeliveries = oneDayTask.deliveries; // ALLDELIVERIES filter by default
+    if (props.completionFilter === "COMPLETED") {
```

Comment:
I see that you declared an enum TaskCompletionOption, it would be better to use the enum when comparing to avoid silly mistakes (you could rename the enum to be exactly what should be displayed in the filter)

Reply:
Updated as suggested :)